### PR TITLE
Ported over necessary example-microservice files

### PR DESCRIPTION
**Pull request by GoodMorningA1i**

### CODE REVIEW

File: typescript/api/api_service_client/api_client.ts [hunk 1]
```diff
@@ -0,0 +1,143 @@
+//"io/ioutil"
+import { Result, ok, err } from "neverthrow";
+import * as httpStatusCode from "http-status-codes"
```
Comment (GoodMorningA1i) [inline typescript/api/api_service_client/api_client.ts:3]: semicolon

File: typescript/api/api_service_client/api_client.ts [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+import * as axios from "axios";
+
+const PERSON_ENDPOINT: string = "person";
+const TEXT_CONTEXT_TYPE: string = "text/plain"; //TODO (Ali)
+
+const TIMEOUT_SECONDS: number = 2; //TODO (Ali)
+const INCREMENT_BOOKS_READ_ENDPOINT: string = "incrementBooksRead";
+
+const HEALTHCHECK_URL_SLUG = "health";
```
Comment (GoodMorningA1i) [inline typescript/api/api_service_client/api_client.ts:12]: type alias

File: typescript/api/api_service_client/api_client.ts [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+const HEALTHY_VALUE: string = "healthy";
+
+class Person {
+	private readonly booksRead: number;
+
+	constructor() {}
+}
+
+class APIClient {
+	//httpClient http.Client
```
Comment (GoodMorningA1i) [inline typescript/api/api_service_client/api_client.ts:22]: add TODO (Ali) here

File: typescript/api/api_service_client/api_client.ts [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+	private readonly ipAddr: string;
+	private readonly port: number;
+
+    constructor (ipAddr: string, port: number) {
```
Comment (GoodMorningA1i) [inline typescript/api/api_service_client/api_client.ts:26]: indenting issues

File: typescript/datastore/datastore_service_client/datastore_client.ts [hunk 5]
```diff
@@ -0,0 +1,160 @@
+import { Result, err, ok } from "neverthrow";
+//"io/ioutil"
+import * as axios from "axios";
+import * as httpStatusCode from "http-status-codes"
```
Comment (GoodMorningA1i) [inline typescript/datastore/datastore_service_client/datastore_client.ts:4]: semicolon

File: typescript/datastore/datastore_service_client/datastore_client.ts [hunk 6]
```diff
@@ -0,0 +1,160 @@
...
+		const url: string = this.getUrlForKey(key);
+		const resp: axios.AxiosResponse<any> = await axios.default.get(url); //TODO (Ali) - might need to catch error to make up for line below
+
+		//TOOD (Ali) - since I removed http.Client struct, I might remove the following error check
+		// if err !== nil {
+		// 	return false, stacktrace.Propagate(err, "An error occurred requesting data for key '%v'", key)
+		// }
+		
+		if (resp.status === httpStatusCode.StatusCodes.OK) {
+			return ok(true)
```
Comment (GoodMorningA1i) [inline typescript/datastore/datastore_service_client/datastore_client.ts:54]: semicolon

File: typescript/api/api_service_client/api_client.ts [hunk 7]
```diff
@@ -0,0 +1,143 @@
...
+		this.ipAddr = ipAddr;
+		this.port = port;
+    }
+
+	public async addPerson(id: number): Promise<Result<null, Error>> {
+		const url: string = this.getPersonUrlForId(id);
+		const resp: axios.AxiosResponse<any> = await axios.default.post(url, null); //TODO (Ali) - might need to catch error to make up for line below ; content type missing in POST request
+		
+		//TOOD (Ali) - since I removed http.Client struct, I might remove the following error check
+		// if err != nil {
```
Comment (GoodMorningA1i) [inline typescript/api/api_service_client/api_client.ts:36]: change to !==

File: typescript/api/api_service_client/api_client.ts [hunk 8]
```diff
@@ -0,0 +1,143 @@
...
+		}
+		return ok(null);
+	}
+
+	public async getPerson(id: number): Promise<Result<Person, Error>> {
+		const url: string = this.getPersonUrlForId(id);
+		const resp: axios.AxiosResponse<any> = await axios.default.get(url); //TODO (Ali) - might need to catch error to make up for line below
+		
+		//TOOD (Ali) - since I removed http.Client struct, I might remove the following error check
+		// if err != nil {
```
Comment (GoodMorningA1i) [inline typescript/api/api_service_client/api_client.ts:50]: !==

File: typescript/api/api_service_client/api_client.ts [hunk 9]
```diff
@@ -0,0 +1,143 @@
...
+		// }
+		if (resp.status !== httpStatusCode.StatusCodes.OK) {
+			return err(new Error("Getting person with ID '" + id + "' returned non-OK status code " + resp.status + ""));
+		}
+		const body: any = resp.data; //TODO (Ali) - building this for JSON.parse
+
+		//TODO (Ali) - how do I deal with a response type of <any>, I can't guarantee on it
+		// defer body.Close()
+		// bodyBytes, err := ioutil.ReadAll(body)
+		// if err != nil {
```
Comment (GoodMorningA1i) [inline typescript/api/api_service_client/api_client.ts:61]: !==

File: typescript/api/api_service_client/api_client.ts [hunk 10]
```diff
@@ -0,0 +1,143 @@
...
+		
+		return ok(person);
+	}
+
+	public async incrementBooksRead(id: number): Promise<Result<null, Error>> {
+		const url: string = "http://" + this.ipAddr + ":" + this.port + "/"+ INCREMENT_BOOKS_READ_ENDPOINT +"/" + id + ""; //TODO (Ali) - shouldn't we be calling getPersonUrlForId
+		const resp: axios.AxiosResponse<any> = await axios.default.post(url, null); //TODO (Ali) - might need to catch error to make up for line below ; content type missing in POST request
+		
+		//TOOD (Ali) - since I removed http.Client struct, I might remove the following error check
+		// if err != nil {
```
Comment (GoodMorningA1i) [inline typescript/api/api_service_client/api_client.ts:80]: !==

File: typescript/datastore/datastore_service_client/datastore_client.ts [hunk 11]
```diff
@@ -0,0 +1,160 @@
...
+		const resp: axios.AxiosResponse<any> = await axios.default.get(url); //TODO (Ali) - might need to do catch error to make up for line below
+		
+		//TOOD (Ali) - since I removed http.Client struct, I might remove the following error check
+		// if err !== nil {
+		// 	return "", stacktrace.Propagate(err, "An error occurred requesting data for key '%v'", key)
+		// }
+		if (resp.status !== httpStatusCode.StatusCodes.OK) {
+			return err(new Error("A non-" + resp.status + " status code was returned"));
+		}
+		const body: any = resp.data
```
Comment (GoodMorningA1i) [inline typescript/datastore/datastore_service_client/datastore_client.ts:76]: semicolon

File: typescript/datastore/datastore_service_client/datastore_client.ts [hunk 12]
```diff
@@ -0,0 +1,160 @@
...
+		// if err !== nil {
+		// 	return stacktrace.Propagate(err, "An error requesting to upsert data '%v' to key '%v'", value, key)
+		// }
+		if (resp.status !== httpStatusCode.StatusCodes.OK) {
+			return err(new Error("A non-" + resp.status + " status code was returned"));
+		}
+		return ok(null);
+	}
+
+	public getUrlForKey(key: string): string {
```
Comment (GoodMorningA1i) [inline typescript/datastore/datastore_service_client/datastore_client.ts:105]: async?

File: typescript/api/api_service_client/api_client.ts [hunk 13]
```diff
@@ -0,0 +1,143 @@
...
+		const bodyStr: string = String(resp.data); //TODO - we might need to keep this as any type, and remove lines below
+
+		if (bodyStr !== HEALTHY_VALUE) {
+			return err(new Error("Expected response body text '" + HEALTHY_VALUE + "' from endpoint '" + url + "' but got '" + bodyStr +  "' instead"));
+		}
+
+		return ok(null);
+	}
+
+	public getPersonUrlForId(id: number): string {
```
Comment (GoodMorningA1i) [inline typescript/api/api_service_client/api_client.ts:127]: async?

File: typescript/api/api_service_client/api_client.ts [hunk 14]
```diff
@@ -0,0 +1,143 @@
...
+	private readonly port: number;
+	
+	constructor (ipAddr: string, port: number) {
+		this.ipAddr = ipAddr;
+		this.port = port;
+	}
+
+	public async addPerson(id: number): Promise<Result<null, Error>> {
+		const url: string = this.getPersonUrlForId(id);
+		const resp: axios.AxiosResponse<any> = await axios.default.post(url, null); //TODO (Ali) - might need to catch error to make up for line below ; content type missing in POST request
```
Comment (mieubrisse) [inline typescript/api/api_service_client/api_client.ts:33]: Yes, you will - rejected promises result in exceptions upon `await`

File: typescript/api/api_service_client/api_client.ts [hunk 15]
```diff
@@ -0,0 +1,143 @@
...
+		
+		//TOOD (Ali) - since I removed http.Client struct, I might remove the following error check
+		// if err !== nil {
```
Comment (mieubrisse) [inline typescript/api/api_service_client/api_client.ts:36]: yep, those are prob just differences in Go vs Typescript

File: typescript/api/api_service_client/api_client.ts [hunk 16]
```diff
@@ -0,0 +1,143 @@
...
+		const resp: axios.AxiosResponse<any> = await axios.default.get(url); //TODO (Ali) - might need to catch error to make up for line below
+		
+		//TOOD (Ali) - since I removed http.Client struct, I might remove the following error check
+		// if err !== nil {
+		// 	return Person{}, stacktrace.Propagate(err, "An error occurred making the request to get person with ID '%v'", id)
+		// }
+		if (resp.status !== httpStatusCode.StatusCodes.OK) {
+			return err(new Error("Getting person with ID '" + id + "' returned non-OK status code " + resp.status + ""));
+		}
+		const body: any = resp.data; //TODO (Ali) - building this for JSON.parse
```
Comment (mieubrisse) [inline typescript/api/api_service_client/api_client.ts:56]: Is this really `any` type? It's not `string`, or `bytes`?

File: typescript/api/api_service_client/api_client.ts [hunk 17]
```diff
@@ -0,0 +1,143 @@
...
+
+		const url: string = "http://"+this.ipAddr+":"+this.port+"/"+HEALTHCHECK_URL_SLUG+"";
+		let respResult: Result<axios.AxiosResponse<any>, Error>;
+
+		for (let i = 0 ; i < retries ; i++) {
+			respResult = await this.makeHttpGetRequest(url);
+			if (respResult.isOk()) {
+				break;
+			}
+			await new Promise(f => setTimeout(f, retriesDelayMilliseconds));
```
Comment (mieubrisse) [inline typescript/api/api_service_client/api_client.ts:102]: `f` should be called `resolve`

File: typescript/api/api_service_client/api_client.ts [hunk 18]
```diff
@@ -0,0 +1,143 @@
...
+		}
+
+		if (!respResult.isOk()){
+			return err(respResult.error);
+		}
+
+		//TODO (Ali) - how do I deal with a response type of <any>, I can't guarantee on it
```
Comment (mieubrisse) [inline typescript/api/api_service_client/api_client.ts:109]: Seems like it should be `string `or `bytes` or something instead

File: typescript/datastore/datastore_service_client/datastore_client.ts [hunk 19]
```diff
@@ -0,0 +1,160 @@
...
+		}
+		const body: any = resp.data;
+
+		//TODO (Ali) - how do I deal with a response type of <any>, I can't guarantee on it
+		// defer body.Close()
+		// bodyBytes, err := ioutil.ReadAll(body)
+		// if err !== nil {
+		// 	return "", stacktrace.Propagate(err, "An error occurred reading the response body")
+		// }
+		return ok(String(body)); //TODO - we might need to keep this as any type
```
Comment (mieubrisse) [inline typescript/datastore/datastore_service_client/datastore_client.ts:84]: mm should def be a `string` return type

File: typescript/package.json [hunk 20]
```diff
@@ -0,0 +1,31 @@
+{
```
Comment (mieubrisse) [inline typescript/package.json:1]: Why is this a new file? Your `GoodMorningAli/porting-typescript...` library should already have a `package.json`; I created it on Friday

**Review by mieubrisse — APPROVED** [review]
Approved pending comments

File: typescript/api/api_service_client/api_client.ts (hunk 16 above)
Comment (GoodMorningA1i) [reply typescript/api/api_service_client/api_client.ts:56]: Oh wait, the documentation said it was <any> but then I found that online it said it returned a string: https://stackoverflow.com/questions/62290771/axios-returns-a-string-instead-of-json-vue-js

File: typescript/package.json (hunk 20 above)
Comment (GoodMorningA1i) [reply typescript/package.json:1]: This is the kurtosis-testsuite-starter-pack repo, you set everything up inside kurtosis-testsuite-api-lib on Friday if I'm not mistaken

**Pull request merged**